Lib/functools.py

```diff
@@ -374,7 +331,49 @@ class partial:
     __slots__ = ("func", "args", "keywords", "_phcount", "_merger",
                  "__dict__", "__weakref__")
 
-    __new__ = _partial_new
+    def __new__(cls, func, /, *args, **keywords):
+        if not callable(func):
+            raise TypeError("the first argument must be callable")
+        if args and args[-1] is Placeholder:
+            # Trim trailing placeholders
+            j = len(args) - 1
+            if not j:
+                args = ()
+            else:
+                while (j := j - 1) >= 0:
+                    if args[j] is not Placeholder:
+                        break
+                args = args[:j + 1]
```

Comment:
This might be a bit simpler:
```suggestion
        # Trim trailing placeholders
        while args and args[-1] is Placeholder:
            args = args[:-1]
```

Reply:
Thanks. I am going to attempt to remove this restriction altogether.